packages/crawler/src/page-processors/page-processor-base.ts

```diff
@@ -14,43 +15,62 @@ export type PartialScanData = {
 } & Partial<ScanData>;
 
 export interface PageProcessor {
-    pageProcessor: Apify.PuppeteerHandlePage;
+    pageHandler: Apify.PuppeteerHandlePage;
     gotoFunction: Apify.PuppeteerGoto;
     pageErrorProcessor: Apify.HandleFailedRequest;
 }
 
 export abstract class PageProcessorBase implements PageProcessor {
+    /**
+     * Timeout in which page navigation needs to finish, in seconds.
+     */
+    public gotoTimeoutSecs = 30;
+
     /**
      * This function is called to extract data from a single web page
      * 'page' is an instance of Puppeteer.Page with page.goto(request.url) already called
      * 'request' is an instance of Request class with information about the page to load
      */
-    public abstract pageProcessor: Apify.PuppeteerHandlePage;
-
-    /**
-     * Timeout in which page navigation needs to finish, in seconds.
-     */
-    public gotoTimeoutSecs = 30;
+    protected abstract processPage: Apify.PuppeteerHandlePage;
 
     public constructor(
+        protected readonly logger: Logger,
         protected readonly requestQueue: Apify.RequestQueue,
         protected readonly discoveryPatterns?: string[],
-        protected readonly accessibilityScanOp: AccessibilityScanOperation = new AccessibilityScanOperation(),
+        protected readonly accessibilityScanOp: AccessibilityScanOperation = new AccessibilityScanOperation(logger),
         protected readonly dataStore: DataStore = new LocalDataStore(scanResultStorageName),
         protected readonly blobStore: BlobStore = new LocalBlobStore(scanResultStorageName),
         private readonly enqueueLinksExt: typeof Apify.utils.enqueueLinks = Apify.utils.enqueueLinks,
         private readonly gotoExtended: typeof Apify.utils.puppeteer.gotoExtended = Apify.utils.puppeteer.gotoExtended,
     ) {}
 
+    public pageHandler: Apify.PuppeteerHandlePage = async (inputs: Apify.PuppeteerHandlePageInputs) => {
+        try {
+            await this.processPage(inputs);
+        } catch (err) {
+            await this.logPageError(inputs.request, err as Error);
+
+            // Throw the error so Apify puts it back into the queue to retry
+            throw err;
+        }
+    };
+
     /**
      * Overrides the function that opens the page in Puppeteer.
      * Return the result of Puppeteer's [page.goto()](https://pptr.dev/#?product=Puppeteer&show=api-pagegotourl-options) function.
      */
     public gotoFunction: Apify.PuppeteerGoto = async (inputs: Apify.PuppeteerGotoInputs) => {
-        return this.gotoExtended(inputs.page, inputs.request, {
-            waitUntil: 'networkidle0',
-            timeout: this.gotoTimeoutSecs * 1000,
-        });
+        try {
+            return await this.gotoExtended(inputs.page, inputs.request, {
+                waitUntil: 'networkidle0',
+                timeout: this.gotoTimeoutSecs * 1000,
```

Comment:
do you think it'd be better if we set time out in the "options: Apify.PuppeteerCrawlerOptions" in "this.crawlerFactory.createPuppeteerCrawler" in crawler-engine.ts?